In [5]:
import mne
import os

import os and mne library to access local file directory directly

# Plotting a random sample (raw)

In [6]:
# Path to raw EEG folder
raw_folder = 'Raw_eeg_only'
raw_files = [f for f in os.listdir(raw_folder) if f.endswith('.raw')]

for file in raw_files:
    file_path = os.path.join(raw_folder, file)
    print(f"\nLoading: {file_path}")
    
    # Load raw data
    raw = mne.io.read_raw_egi(file_path, preload=True)
    
    # Only keep EEG channels
    raw.pick_types(eeg=True, stim=False)  # drop stim channels from view
    
    # Apply bandpass filter (1–45 Hz)
    raw.filter(1., 45., fir_design='firwin')

    # Drop E129 if present
    if 'E129' in raw.ch_names:
        raw.drop_channels(['E129'])

    # Set montage (HydroCel 129)
    montage = mne.channels.make_standard_montage('GSN-HydroCel-129')
    raw.set_montage(montage, on_missing='ignore')

    # Plot 30 EEG channels (just a 5-sec window to test)
    raw.plot(n_channels=30, duration=5.0, scalings='auto', block=True, title=f"{file}")
    
    break  # Just test with the first sample


FileNotFoundError: [WinError 3] The system cannot find the path specified: 'Raw_eeg_only'

# store event handler(epochs) in separate fif files to assess brain activity during different events

In [ ]:
import mne
import os

# Folder containing your .raw files
data_folder = 'Raw_eeg_only'
output_folder = 'Epochs_output'
os.makedirs(output_folder, exist_ok=True)

# Loop through each .raw file
for filename in os.listdir(data_folder):
    if filename.endswith('.raw'):
        filepath = os.path.join(data_folder, filename)
        print(f"\nProcessing: {filename}")

        # Load raw EEG
        raw = mne.io.read_raw_egi(filepath, preload=True)

        # Apply bandpass filter
        raw.filter(1., 45., fir_design='firwin')

        # Set montage (handling E129 separately to avoid errors)
        if 'E129' in raw.ch_names:
             raw.drop_channels(['E129'])

        montage = mne.channels.make_standard_montage('GSN-HydroCel-129')
        raw.set_montage(montage)


        # Extract events from annotations
        events, event_id = mne.events_from_annotations(raw)

        # Filter event IDs to only include cue-related triggers (or choose what you want)
        cue_event_ids = {k: v for k, v in event_id.items() if 'cue' in k}

        if not cue_event_ids:
            print(f"No 'cue' events found in {filename}. Skipping.")
            continue

        # Epoch the data (e.g., -0.2 to 0.8s around event)
        try:
            epochs = mne.Epochs(
                raw,
                events,
                event_id=cue_event_ids,
                tmin=-0.2,
                tmax=0.8,
                baseline=(None, 0),
                preload=True
            )

            # Save each subject’s epochs as .fif
            output_name = filename.replace('.raw', '-epo.fif')
            output_path = os.path.join(output_folder, output_name)
            epochs.save(output_path, overwrite=True)

            print(f"Saved epochs: {output_name}")
        except Exception as e:
            print(f"Error processing {filename}: {e}")



Processing: 02010002erp 20150416 1131.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010002erp 20150416 1131.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 197401  =      0.000 ...   789.604 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass, zero-phase, non-causal bandpass filter:
- Windowed time-domain design (firwin) method
- Hamming window with 0.0194 passband ripple and 53 dB stopband attenuation
- Lower passband edge: 1.00
- Lower transition bandwidth: 1.00 Hz (-6 dB cutoff frequency: 0.50 Hz)
- Upper passband edge: 45.00 Hz
- Upper transition bandwidth: 11.25 Hz (-6 dB cutoff frequency: 50.62 Hz)
- Filter length: 825 samples (3.300 s)



[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010002erp 20150416 1131-epo.fif

Processing: 02010004erp 20141219 1602.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010004erp 20141219 1602.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 195027  =      0.000 ...   780.108 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.2s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'Cat1', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010004erp 20141219 1602-epo.fif

Processing: 02010005erp 20150507 0938.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010005erp 20150507 0938.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 232938  =      0.000 ...   931.752 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a o

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.8s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.5s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010005erp 20150507 0938-epo.fif

Processing: 02010006erp 20150528 1007.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010006erp 20150528 1007.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 195040  =      0.000 ...   780.160 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.4s finished


Used Annotations descriptions: ['CELL', 'D127', 'D255', 'DI18', 'DI26', 'DI95', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010006erp 20150528 1007-epo.fif

Processing: 02010008_erp(n) 20150619 1709.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010008_erp(n) 20150619 1709.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 201090  =      0.000 ...   804.360 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parame

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.4s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010008_erp(n) 20150619 1709-epo.fif

Processing: 02010010erp 20150624 1508.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010010erp 20150624 1508.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 216089  =      0.000 ...   864.356 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-p

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'D127', 'DI18', 'DI26', 'DI95', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010010erp 20150624 1508-epo.fif

Processing: 02010011erp 20150625 1545.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010011erp 20150625 1545.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 214203  =      0.000 ...   856.812 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
-----------

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010011erp 20150625 1545-epo.fif

Processing: 02010012erp 20150626 1059.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010012erp 20150626 1059.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 203614  =      0.000 ...   814.456 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.4s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010012erp 20150626 1059-epo.fif

Processing: 02010013erp 20150703 1353.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010013erp 20150703 1353.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 210765  =      0.000 ...   843.060 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.4s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010013erp 20150703 1353-epo.fif

Processing: 02010015erp 20150709 1534.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010015erp 20150709 1534.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 201902  =      0.000 ...   807.608 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.4s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010015erp 20150709 1534-epo.fif

Processing: 02010016erp 20150710 1329.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010016erp 20150710 1329.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 201638  =      0.000 ...   806.552 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010016erp 20150710 1329-epo.fif

Processing: 02010018erp 20150716 1310.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010018erp 20150716 1310.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 201301  =      0.000 ...   805.204 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']


[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010018erp 20150716 1310-epo.fif

Processing: 02010019erp 20150716 1544.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010019erp 20150716 1544.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 196038  =      0.000 ...   784.152 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass, zero-phase, non-causal bandpass filter:
- Windowed time-domain design (firwin) method
- Hamming window with 0.0194 passband ripple and 53 dB stopband attenuation
- Lower passband edge: 1.00
- Lower t

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010019erp 20150716 1544-epo.fif

Processing: 02010021erp 20150805 1818.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010021erp 20150805 1818.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 205092  =      0.000 ...   820.368 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010021erp 20150805 1818-epo.fif

Processing: 02010022erp 20150724 1457.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010022erp 20150724 1457.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 201150  =      0.000 ...   804.600 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']


[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.2s finished


Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010022erp 20150724 1457-epo.fif

Processing: 02010023erp 20150729 1955.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010023erp 20150729 1955.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 211052  =      0.000 ...   844.208 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass, zero-phase, non-causal bandpass filter:
- Windowed time-domain design (firwin) method
- Hamming window with 0.0194 passband ripple and 53 dB stopband attenuation
- Lower passband edge: 1.00
- Lower t

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.8s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.5s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010023erp 20150729 1955-epo.fif

Processing: 02010024erp 20150814 1523.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010024erp 20150814 1523.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 198475  =      0.000 ...   793.900 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.2s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010024erp 20150814 1523-epo.fif

Processing: 02010025erp 20160311 1225.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010025erp 20160311 1225.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 205878  =      0.000 ...   823.512 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010025erp 20160311 1225-epo.fif

Processing: 02010026erp 20160311 1441.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010026erp 20160311 1441.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 238766  =      0.000 ...   955.064 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.8s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.5s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010026erp 20160311 1441-epo.fif

Processing: 02010028erp 20160317 1554.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010028erp 20160317 1554.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 195316  =      0.000 ...   781.264 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.2s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010028erp 20160317 1554-epo.fif

Processing: 02010030erp 20160324 0915.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010030erp 20160324 0915.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 205251  =      0.000 ...   821.004 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.4s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010030erp 20160324 0915-epo.fif

Processing: 02010033erp 20160331 1307.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010033erp 20160331 1307.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 200790  =      0.000 ...   803.160 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.4s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010033erp 20160331 1307-epo.fif

Processing: 02010034erp 20160407 0959.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010034erp 20160407 0959.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 289466  =      0.000 ...  1157.864 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.2s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    1.0s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.9s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010034erp 20160407 0959-epo.fif

Processing: 02010036erp 20160408 1452.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02010036erp 20160408 1452.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 265003  =      0.000 ...  1060.012 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.2s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.9s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.7s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02010036erp 20160408 1452-epo.fif

Processing: 02020008_erp 20150624 1740.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020008_erp 20150624 1740.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 208802  =      0.000 ...   835.208 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pas

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']


[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.5s finished


Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020008_erp 20150624 1740-epo.fif

Processing: 02020010_erp 20150625 1244.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020010_erp 20150625 1244.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 216153  =      0.000 ...   864.612 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass, zero-phase, non-causal bandpass filter:
- Windowed time-domain design (firwin) method
- Hamming window with 0.0194 passband ripple and 53 dB stopband attenuation
- Lower passband edge: 1.00
- Lowe

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020010_erp 20150625 1244-epo.fif

Processing: 02020013_erp 20150629 1622.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020013_erp 20150629 1622.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 203564  =      0.000 ...   814.256 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pa

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.2s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020013_erp 20150629 1622-epo.fif

Processing: 02020014_erp 20150630 1052.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020014_erp 20150630 1052.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 200778  =      0.000 ...   803.112 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pa

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.4s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020014_erp 20150630 1052-epo.fif

Processing: 02020015_erp 20150630 1547.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020015_erp 20150630 1547.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 194990  =      0.000 ...   779.960 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pa

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020015_erp 20150630 1547-epo.fif

Processing: 02020016_erp 20150701 1054.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020016_erp 20150701 1054.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 198414  =      0.000 ...   793.656 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pa

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020016_erp 20150701 1054-epo.fif

Processing: 02020018_erp 20150702 1721.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020018_erp 20150702 1721.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 201202  =      0.000 ...   804.808 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pa

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']


[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.2s finished


Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020018_erp 20150702 1721-epo.fif

Processing: 02020019_erp 20150703 1052.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020019_erp 20150703 1052.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 215166  =      0.000 ...   860.664 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass, zero-phase, non-causal bandpass filter:
- Windowed time-domain design (firwin) method
- Hamming window with 0.0194 passband ripple and 53 dB stopband attenuation
- Lower passband edge: 1.00
- Lowe

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.2s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.8s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.4s finished


Used Annotations descriptions: ['CELL', 'D127', 'DI18', 'DI26', 'DI95', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020019_erp 20150703 1052-epo.fif

Processing: 02020020_erp 20150703 1810.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020020_erp 20150703 1810.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 195676  =      0.000 ...   782.704 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
--------

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.2s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020020_erp 20150703 1810-epo.fif

Processing: 02020021_erp 20150707 1751.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020021_erp 20150707 1751.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 201153  =      0.000 ...   804.612 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pa

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']


[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020021_erp 20150707 1751-epo.fif

Processing: 02020022_erp 20150707 1515.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020022_erp 20150707 1515.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 201588  =      0.000 ...   806.352 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass, zero-phase, non-causal bandpass filter:
- Windowed time-domain design (firwin) method
- Hamming window with 0.0194 passband ripple and 53 dB stopband attenuation
- Lower passband edge: 1.00
- Lowe

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.2s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020022_erp 20150707 1515-epo.fif

Processing: 02020023_erpnew 20150709 110.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020023_erpnew 20150709 110.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 201651  =      0.000 ...   806.604 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a on

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020023_erpnew 20150709 110-epo.fif

Processing: 02020025_erp 20150713 1541.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020025_erp 20150713 1541.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 197977  =      0.000 ...   791.908 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020025_erp 20150713 1541-epo.fif

Processing: 02020026_erp 20150714 1428.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020026_erp 20150714 1428.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 204252  =      0.000 ...   817.008 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pa

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020026_erp 20150714 1428-epo.fif

Processing: 02020027_erp 20150713 1116.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020027_erp 20150713 1116.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 197814  =      0.000 ...   791.256 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pa

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.2s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020027_erp 20150713 1116-epo.fif

Processing: 02020029_erp 20150715 1401.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02020029_erp 20150715 1401.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 206664  =      0.000 ...   826.656 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pa

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.4s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02020029_erp 20150715 1401-epo.fif

Processing: 02030002erp_new 20151022 14.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02030002erp_new 20151022 14.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 196128  =      0.000 ...   784.512 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02030002erp_new 20151022 14-epo.fif

Processing: 02030003erp 20151022 1212.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02030003erp 20151022 1212.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 200676  =      0.000 ...   802.704 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pas

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.2s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02030003erp 20151022 1212-epo.fif

Processing: 02030004erp 20151026 2004.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02030004erp 20151026 2004.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 199339  =      0.000 ...   797.356 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02030004erp 20151026 2004-epo.fif

Processing: 02030005erp 20151026 2116.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02030005erp 20151026 2116.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 202702  =      0.000 ...   810.808 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass,

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']


[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.2s finished


Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02030005erp 20151026 2116-epo.fif

Processing: 02030006 20151103 1801.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02030006 20151103 1801.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 196063  =      0.000 ...   784.252 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass, zero-phase, non-causal bandpass filter:
- Windowed time-domain design (firwin) method
- Hamming window with 0.0194 passband ripple and 53 dB stopband attenuation
- Lower passband edge: 1.00
- Lower transit

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.2s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02030006 20151103 1801-epo.fif

Processing: 02030007 20151103 2129.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02030007 20151103 2129.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 205139  =      0.000 ...   820.556 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass, zero-pha

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02030007 20151103 2129-epo.fif

Processing: 02030009erp 20151105 1207.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02030009erp 20151105 1207.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 198224  =      0.000 ...   792.896 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass, ze

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']


[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.2s finished


Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02030009erp 20151105 1207-epo.fif

Processing: 02030014erp 20151117 1419.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02030014erp 20151117 1419.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 202638  =      0.000 ...   810.552 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass, zero-phase, non-causal bandpass filter:
- Windowed time-domain design (firwin) method
- Hamming window with 0.0194 passband ripple and 53 dB stopband attenuation
- Lower passband edge: 1.00
- Lower t

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']


[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.2s finished


Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02030014erp 20151117 1419-epo.fif

Processing: 02030017erp 20151208 1351.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02030017erp 20151208 1351.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 208290  =      0.000 ...   833.160 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass, zero-phase, non-causal bandpass filter:
- Windowed time-domain design (firwin) method
- Hamming window with 0.0194 passband ripple and 53 dB stopband attenuation
- Lower passband edge: 1.00
- Lower t

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.6s


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']


[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.2s finished


Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02030017erp 20151208 1351-epo.fif

Processing: 02030018erp 20151208 1514.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02030018erp 20151208 1514.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 212138  =      0.000 ...   848.552 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass, zero-phase, non-causal bandpass filter:
- Windowed time-domain design (firwin) method
- Hamming window with 0.0194 passband ripple and 53 dB stopband attenuation
- Lower passband edge: 1.00
- Lower t

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.4s finished


Used Annotations descriptions: ['CELL', 'D127', 'DI18', 'DI26', 'DI27', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02030018erp 20151208 1514-epo.fif

Processing: 02030019_erp 20151230 1331.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02030019_erp 20151230 1331.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 204702  =      0.000 ...   818.808 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']


[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02030019_erp 20151230 1331-epo.fif

Processing: 02030020_erp 20151230 1443.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02030020_erp 20151230 1443.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 197627  =      0.000 ...   790.508 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass, zero-phase, non-causal bandpass filter:
- Windowed time-domain design (firwin) method
- Hamming window with 0.0194 passband ripple and 53 dB stopband attenuation
- Lower passband edge: 1.00
- Lowe

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s


Used Annotations descriptions: ['CELL', 'D127', 'DI26', 'DI27', 'DI95', 'DIN2', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']


[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02030020_erp 20151230 1443-epo.fif

Processing: 02030021erp 20160105 1204.raw
Reading EGI header from c:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Raw_eeg_only\02030021erp 20160105 1204.raw...
    Reading events ...
    Assembling measurement info ...
    Excluding events {} ...
Reading 0 ... 197977  =      0.000 ...   791.908 secs...
Filtering raw data in 1 contiguous segment
Setting up band-pass filter from 1 - 45 Hz

FIR filter parameters
---------------------
Designing a one-pass, zero-phase, non-causal bandpass filter:
- Windowed time-domain design (firwin) method
- Hamming window with 0.0194 passband ripple and 53 dB stopband attenuation
- Lower passband edge: 1.00
- Lower 

[Parallel(n_jobs=1)]: Done  17 tasks      | elapsed:    0.1s
[Parallel(n_jobs=1)]: Done  71 tasks      | elapsed:    0.7s
[Parallel(n_jobs=1)]: Done 129 out of 129 | elapsed:    1.3s finished


Used Annotations descriptions: ['CELL', 'SESS', 'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp', 'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp']
Not setting metadata
480 matching events found
Setting baseline interval to [-0.2, 0.0] s
Applying baseline correction (mode: mean)
0 projection items activated
Using data from preloaded Raw for 480 events and 251 original time points ...
0 bad epochs dropped
Saved epochs: 02030021erp 20160105 1204-epo.fif


# Prepping data for eegnet model (input specific for model shape and requirements)

## NOTE : Used a standard fif file (02010006) as a reference to clean up other files(it failed) 

In [ ]:
import os
import mne

folder = r"C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Epochs_output"

max_channels = 0
ref_epochs = None
ref_file = ""

for fname in os.listdir(folder):
    if fname.endswith("-epo.fif"):
        fpath = os.path.join(folder, fname)
        try:
            epochs = mne.read_epochs(fpath, preload=False, verbose=False)
            n_ch = len(epochs.ch_names)
            print(f"📄 {fname} → {n_ch} channels")

            if n_ch > max_channels:
                max_channels = n_ch
                ref_epochs = epochs
                ref_file = fname
        except Exception as e:
            print(f"❌ {fname} skipped due to error: {e}")

if ref_epochs:
    print(f"\n✅ Using reference file: {ref_file} with {max_channels} channels")
    ref_channel_order = ref_epochs.ch_names
else:
    print("🚨 No valid .epo.fif file found!")


📄 02010002erp 20150416 1131-epo.fif → 149 channels

📄 02010004erp 20141219 1602-epo.fif → 150 channels
📄 02010005erp 20150507 0938-epo.fif → 149 channels
📄 02010004erp 20141219 1602-epo.fif → 150 channels
📄 02010005erp 20150507 0938-epo.fif → 149 channels
📄 02010006erp 20150528 1007-epo.fif → 154 channels
📄 02010008_erp(n) 20150619 1709-epo.fif → 149 channels
📄 02010006erp 20150528 1007-epo.fif → 154 channels
📄 02010008_erp(n) 20150619 1709-epo.fif → 149 channels
📄 02010010erp 20150624 1508-epo.fif → 153 channels
📄 02010011erp 20150625 1545-epo.fif → 149 channels
📄 02010010erp 20150624 1508-epo.fif → 153 channels
📄 02010011erp 20150625 1545-epo.fif → 149 channels
📄 02010012erp 20150626 1059-epo.fif → 149 channels
📄 02010013erp 20150703 1353-epo.fif → 149 channels
📄 02010012erp 20150626 1059-epo.fif → 149 channels
📄 02010013erp 20150703 1353-epo.fif → 149 channels
📄 02010015erp 20150709 1534-epo.fif → 149 channels
📄 02010016erp 20150710 1329-epo.fif → 149 channels
📄 02010015erp 20150709

### ALIGNING ALL SAMPLES  TO HAVE UNIFIED CHANNEL NUMBER ( From N-> 149 -> 128)


In [ ]:
import mne
import os

input_folder = r"C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Epochs_output"
output_folder = os.path.join(input_folder, "Epochs_aligned")
os.makedirs(output_folder, exist_ok=True)

# Use previously found values
reference_file = "02010006erp 20150528 1007-epo.fif"
ref_channel_order = ref_epochs.ch_names

for fname in os.listdir(input_folder):
    if fname.endswith("-epo.fif") and fname != reference_file:
        fpath = os.path.join(input_folder, fname)
        try:
            epochs = mne.read_epochs(fpath, preload=True, verbose=False)
            current_chs = set(epochs.ch_names)
            ref_chs = set(ref_channel_order)

            # Drop extra channels not in reference
            extra_chs = current_chs - ref_chs
            if extra_chs:
                print(f"➖ Dropping {len(extra_chs)} extra channels in {fname}")
                epochs.drop_channels(list(extra_chs))

            # Reorder to match reference
            intersecting_chs = [ch for ch in ref_channel_order if ch in epochs.ch_names]
            epochs.reorder_channels(intersecting_chs)

            # Save aligned
            out_path = os.path.join(output_folder, fname)
            epochs.save(out_path, overwrite=True)
            print(f"✅ Saved aligned: {fname}")

        except Exception as e:
            print(f"❌ Failed processing {fname}: {e}")


✅ Saved aligned: 02010002erp 20150416 1131-epo.fif
➖ Dropping 1 extra channels in 02010004erp 20141219 1602-epo.fif
➖ Dropping 1 extra channels in 02010004erp 20141219 1602-epo.fif
✅ Saved aligned: 02010004erp 20141219 1602-epo.fif
✅ Saved aligned: 02010004erp 20141219 1602-epo.fif
✅ Saved aligned: 02010005erp 20150507 0938-epo.fif
✅ Saved aligned: 02010005erp 20150507 0938-epo.fif
✅ Saved aligned: 02010008_erp(n) 20150619 1709-epo.fif
✅ Saved aligned: 02010008_erp(n) 20150619 1709-epo.fif
✅ Saved aligned: 02010010erp 20150624 1508-epo.fif
✅ Saved aligned: 02010010erp 20150624 1508-epo.fif
✅ Saved aligned: 02010011erp 20150625 1545-epo.fif
✅ Saved aligned: 02010011erp 20150625 1545-epo.fif
✅ Saved aligned: 02010012erp 20150626 1059-epo.fif
✅ Saved aligned: 02010012erp 20150626 1059-epo.fif
✅ Saved aligned: 02010013erp 20150703 1353-epo.fif
✅ Saved aligned: 02010013erp 20150703 1353-epo.fif
✅ Saved aligned: 02010015erp 20150709 1534-epo.fif
✅ Saved aligned: 02010015erp 20150709 1534-epo

In [ ]:
import mne
import os

input_folder = r"C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Epochs_output"
aligned_folder = os.path.join(input_folder, "Epochs_aligned")
os.makedirs(aligned_folder, exist_ok=True)

# Manually copied reference file (already inside aligned_folder)
ref_path = os.path.join(aligned_folder, "02010006erp 20150528 1007-epo.fif")
ref_epochs = mne.read_epochs(ref_path, preload=False)
ref_channel_order = ref_epochs.ch_names

for fname in os.listdir(input_folder):
    if fname.endswith("-epo.fif") and fname != "02010006erp 20150528 1007-epo.fif":
        input_path = os.path.join(input_folder, fname)
        output_path = os.path.join(aligned_folder, fname)

        try:
            print(f"\n📂 Processing {fname}...")
            epochs = mne.read_epochs(input_path, preload=True, verbose=False)
            current_chs = set(epochs.ch_names)
            ref_chs = set(ref_channel_order)

            # Drop extra channels not in reference
            extra_chs = current_chs - ref_chs
            if extra_chs:
                print(f"➖ Dropping {len(extra_chs)} extra channels: {extra_chs}")
                epochs.drop_channels(list(extra_chs))

            # Reorder channels to match reference
            intersecting_chs = [ch for ch in ref_channel_order if ch in epochs.ch_names]
            epochs.reorder_channels(intersecting_chs)

            # Save aligned file
            epochs.save(output_path, overwrite=True)
            print(f"✅ Saved aligned file to: {output_path}")

        except Exception as e:
            print(f"❌ Error processing {fname}: {e}")


Reading C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Epochs_output\Epochs_aligned\02010006erp 20150528 1007-epo.fif ...
    Found the data of interest:
        t =    -200.00 ...     800.00 ms
        0 CTF compensation matrices available
Not setting metadata
480 matching events found
No baseline correction applied
0 projection items activated

📂 Processing 02010002erp 20150416 1131-epo.fif...
Overwriting existing file.
Overwriting existing file.
✅ Saved aligned file to: C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Epochs_output\Epochs_aligned\02010002erp 20150416 1131-epo.fif

📂 Processing 02010004erp 20141219 1602-epo.fif...
➖ Dropping 1 extra channels: {'Cat1'}
Overwriting existing file.
Overwriting existing file.
✅ Saved aligned file to: C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Epochs_output\Epochs_aligned\02010004erp 20141219 1602-epo.fif

📂 Processing 02010005erp 

#### Checking Shape ( X,Y,Z)  of each files in the Aligned folder 

In [ ]:
import mne
from pathlib import Path

# Set the aligned folder path
aligned_dir = Path(r"C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Epochs_output\Epochs_aligned")

# Reference file for checks
ref_path = aligned_dir / "02010006erp 20150528 1007-epo.fif"
ref_epochs = mne.read_epochs(ref_path, preload=False)
ref_ch_names = ref_epochs.ch_names
ref_shape = ref_epochs.get_data().shape

print(f"\n🔍 Reference file: {ref_path.name}")
print(f"📏 Expected shape: {ref_shape}")
print(f"🧠 Expected channels: {len(ref_ch_names)} channels\n")

# Loop through all aligned .fif files
for fpath in sorted(aligned_dir.glob("*-epo.fif")):
    try:
        epochs = mne.read_epochs(fpath, preload=False, verbose=False)

        # Check shape
        shape_match = epochs.get_data().shape == ref_shape

        # Check number of channels
        ch_count_match = len(epochs.ch_names) == len(ref_ch_names)

        # Check channel names and order
        ch_names_match = epochs.ch_names == ref_ch_names

        if shape_match and ch_count_match and ch_names_match:
            print(f"✅ {fpath.name} passed all checks.")
        else:
            print(f"⚠️ {fpath.name} failed:")
            if not shape_match:
                print(f"   - Shape mismatch: {epochs.get_data().shape}")
            if not ch_count_match:
                print(f"   - Channel count mismatch: {len(epochs.ch_names)}")
            if not ch_names_match:
                print(f"   - Channel names/order mismatch")

    except Exception as e:
        print(f"❌ Error loading {fpath.name}: {e}")


Reading C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Epochs_output\Epochs_aligned\02010006erp 20150528 1007-epo.fif ...
    Found the data of interest:
        t =    -200.00 ...     800.00 ms
        0 CTF compensation matrices available
Not setting metadata
480 matching events found
No baseline correction applied
0 projection items activated
Loading data for 480 events and 251 original time points ...

🔍 Reference file: 02010006erp 20150528 1007-epo.fif
📏 Expected shape: (480, 154, 251)
🧠 Expected channels: 154 channels

Loading data for 480 events and 251 original time points ...
⚠️ 02010002erp 20150416 1131-epo.fif failed:
Loading data for 480 events and 251 original time points ...
   - Shape mismatch: (480, 149, 251)
   - Channel count mismatch: 149
   - Channel names/order mismatch
Loading data for 480 events and 251 original time points ...
⚠️ 02010004erp 20141219 1602-epo.fif failed:
Loading data for 480 events and 251 original time points

#### Narrowing down to 149 channels per file 

In [ ]:
import os
import mne

aligned_folder = r"C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Epochs_output\Epochs_aligned"
cleaned_folder = "Cleaned_epochs/"
os.makedirs(cleaned_folder, exist_ok=True)

# This is your final clean reference channel list
common_channels = [  
    'CELL', 'E1', 'E10', 'E100', 'E101', 'E102', 'E103', 'E104', 'E105', 'E106', 'E107', 'E108', 'E109',
    'E11', 'E110', 'E111', 'E112', 'E113', 'E114', 'E115', 'E116', 'E117', 'E118', 'E119', 'E12', 'E120',
    'E121', 'E122', 'E123', 'E124', 'E125', 'E126', 'E127', 'E128', 'E13', 'E14', 'E15', 'E16', 'E17', 'E18',
    'E19', 'E2', 'E20', 'E21', 'E22', 'E23', 'E24', 'E25', 'E26', 'E27', 'E28', 'E29', 'E3', 'E30', 'E31',
    'E32', 'E33', 'E34', 'E35', 'E36', 'E37', 'E38', 'E39', 'E4', 'E40', 'E41', 'E42', 'E43', 'E44', 'E45',
    'E46', 'E47', 'E48', 'E49', 'E5', 'E50', 'E51', 'E52', 'E53', 'E54', 'E55', 'E56', 'E57', 'E58', 'E59',
    'E6', 'E60', 'E61', 'E62', 'E63', 'E64', 'E65', 'E66', 'E67', 'E68', 'E69', 'E7', 'E70', 'E71', 'E72',
    'E73', 'E74', 'E75', 'E76', 'E77', 'E78', 'E79', 'E8', 'E80', 'E81', 'E82', 'E83', 'E84', 'E85', 'E86',
    'E87', 'E88', 'E89', 'E9', 'E90', 'E91', 'E92', 'E93', 'E94', 'E95', 'E96', 'E97', 'E98', 'E99', 'SESS',
    'TRSP', 'bgin', 'epoc', 'fcue', 'fdot', 'ffix', 'fisi', 'fwrp', 'hcue', 'hdot', 'hfix', 'hisi', 'hwrp',
    'resp', 'scue', 'sdot', 'sfix', 'sisi', 'swrp'
]

for fname in os.listdir(aligned_folder):
    if fname.endswith("-epo.fif"):
        fpath = os.path.join(aligned_folder, fname)
        try:
            epochs = mne.read_epochs(fpath, preload=True, verbose=False)
            keep_ch = [ch for ch in common_channels if ch in epochs.ch_names]
            epochs.pick_channels(keep_ch)
            save_path = os.path.join(cleaned_folder, fname)
            epochs.save(save_path, overwrite=True)
            print(f"✅ Cleaned and saved: {fname}")
        except Exception as e:
            print(f"❌ Failed: {fname} — {e}")


NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Cleaned and saved: 02010002erp 20150416 1131-epo.fif
NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Cleaned and saved: 02010004erp 20141219 1602-epo.fif
NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Cleaned and saved: 02010005erp 20150507 0938-epo.fif
NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Cleaned and saved: 02010006erp 20150528 1007-epo.fif
NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Cleaned and saved: 02010008_erp(n) 20150619 1709-epo.fif
NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Cleaned and saved: 02010010erp 20150624 1508-epo.fif
NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Cleaned and saved: 02010011erp 20150625 1545-epo.fif
NOTE: pick_channels() is a legacy function. New cod

#### Check if all files in cleaned folder have 149 channels

In [ ]:
import os
import mne

all_channels_dict = {}

aligned_folder = r"C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Cleaned_epochs"
for fname in os.listdir(aligned_folder):
    if fname.endswith("-epo.fif"):
        fpath = os.path.join(aligned_folder, fname)
        try:
            epochs = mne.read_epochs(fpath, preload=False, verbose=False)
            all_channels_dict[fname] = set(epochs.ch_names)
            print(f"✅ {fname} → {len(epochs.ch_names)} channels")
        except Exception as e:
            print(f"❌ {fname} failed: {e}")

# Optional: Print all channel names file-by-file
print("\n🔍 Channel names per file:")
for fname, ch_set in all_channels_dict.items():
    print(f"\n📁 {fname}:\n{sorted(ch_set)}")
    


✅ 02010002erp 20150416 1131-epo.fif → 149 channels
✅ 02010004erp 20141219 1602-epo.fif → 149 channels
✅ 02010005erp 20150507 0938-epo.fif → 149 channels
✅ 02010006erp 20150528 1007-epo.fif → 149 channels
✅ 02010008_erp(n) 20150619 1709-epo.fif → 149 channels
✅ 02010010erp 20150624 1508-epo.fif → 149 channels
✅ 02010011erp 20150625 1545-epo.fif → 149 channels
✅ 02010012erp 20150626 1059-epo.fif → 149 channels
✅ 02010013erp 20150703 1353-epo.fif → 149 channels
✅ 02010015erp 20150709 1534-epo.fif → 149 channels
✅ 02010016erp 20150710 1329-epo.fif → 149 channels
✅ 02010018erp 20150716 1310-epo.fif → 149 channels
✅ 02010019erp 20150716 1544-epo.fif → 149 channels
✅ 02010021erp 20150805 1818-epo.fif → 149 channels
✅ 02010022erp 20150724 1457-epo.fif → 149 channels
✅ 02010023erp 20150729 1955-epo.fif → 149 channels
✅ 02010024erp 20150814 1523-epo.fif → 149 channels
✅ 02010025erp 20160311 1225-epo.fif → 149 channels
✅ 02010026erp 20160311 1441-epo.fif → 149 channels
✅ 02010028erp 20160317 1554

In [ ]:
import os
import mne

# Paths
input_dir = r"C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Cleaned_epochs"
output_dir = "filtered128_folder"
os.makedirs(output_dir, exist_ok=True)

# Define desired 128 EEG channels
eeg128_channels = [f"E{i}" for i in range(1, 129)]

# Loop through all files
for filename in os.listdir(input_dir):
    if filename.endswith(".fif"):
        fpath = os.path.join(input_dir, filename)
        try:
            raw = mne.read_epochs(fpath, preload=True, verbose=False)
            existing_channels = [ch for ch in eeg128_channels if ch in raw.ch_names]
            raw.pick_channels(existing_channels)
            save_path = os.path.join(output_dir, filename)
            raw.save(save_path, overwrite=True)
            print(f"✅ Filtered {filename}")
        except Exception as e:
            print(f"❌ Failed {filename}: {e}")


NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Filtered 02010002erp 20150416 1131-epo.fif
NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Filtered 02010004erp 20141219 1602-epo.fif
NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Filtered 02010005erp 20150507 0938-epo.fif
NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Filtered 02010006erp 20150528 1007-epo.fif
NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Filtered 02010008_erp(n) 20150619 1709-epo.fif
NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Filtered 02010010erp 20150624 1508-epo.fif
NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Filtered 02010011erp 20150625 1545-epo.fif
NOTE: pick_channels() is a legacy function. New code should use inst.pick(...).
✅ Filtered 02010012erp 20150626 1059-epo.

#### Checking if all files in filtered_128 folder have 128 channels 


In [ ]:
import os
import mne

all_channels_dict = {}

aligned_folder = r"C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\filtered128_folder"
for fname in os.listdir(aligned_folder):
    if fname.endswith("-epo.fif"):
        fpath = os.path.join(aligned_folder, fname)
        try:
            epochs = mne.read_epochs(fpath, preload=False, verbose=False)
            all_channels_dict[fname] = set(epochs.ch_names)
            print(f"✅ {fname} → {len(epochs.ch_names)} channels")
        except Exception as e:
            print(f"❌ {fname} failed: {e}")

# Optional: Print all channel names file-by-file
print("\n🔍 Channel names per file:")
for fname, ch_set in all_channels_dict.items():
    print(f"\n📁 {fname}:\n{sorted(ch_set)}")
    


✅ 02010002erp 20150416 1131-epo.fif → 128 channels
✅ 02010004erp 20141219 1602-epo.fif → 128 channels
✅ 02010005erp 20150507 0938-epo.fif → 128 channels
✅ 02010006erp 20150528 1007-epo.fif → 128 channels
✅ 02010008_erp(n) 20150619 1709-epo.fif → 128 channels
✅ 02010010erp 20150624 1508-epo.fif → 128 channels
✅ 02010011erp 20150625 1545-epo.fif → 128 channels
✅ 02010012erp 20150626 1059-epo.fif → 128 channels
✅ 02010013erp 20150703 1353-epo.fif → 128 channels
✅ 02010015erp 20150709 1534-epo.fif → 128 channels
✅ 02010016erp 20150710 1329-epo.fif → 128 channels
✅ 02010018erp 20150716 1310-epo.fif → 128 channels
✅ 02010019erp 20150716 1544-epo.fif → 128 channels
✅ 02010021erp 20150805 1818-epo.fif → 128 channels
✅ 02010022erp 20150724 1457-epo.fif → 128 channels
✅ 02010023erp 20150729 1955-epo.fif → 128 channels
✅ 02010024erp 20150814 1523-epo.fif → 128 channels
✅ 02010025erp 20160311 1225-epo.fif → 128 channels
✅ 02010026erp 20160311 1441-epo.fif → 128 channels
✅ 02010028erp 20160317 1554

#### Since E1-E128 are not in sorted order , sorting channnel number in place 

In [ ]:
import os
import re
import mne
from natsort import natsorted

# Path to your folder
folder_path =r"C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\filtered128_folder"

# Natural sort helper
def natural_sort(ch_names):
    return natsorted(ch_names, key=lambda x: int(re.search(r'\d+', x).group()))

# Iterate over all FIF files and sort channels
for file in os.listdir(folder_path):
    if file.endswith(".fif"):
        file_path = os.path.join(folder_path, file)
        raw = mne.read_epochs(file_path, preload=True)

        # Natural sort channel names
        sorted_ch_names = natural_sort(raw.ch_names)

        # Reorder channels
        raw.reorder_channels(sorted_ch_names)

        # Optional: save back
        raw.save(file_path, overwrite=True)

        print(f"{file}: {raw.ch_names}")  # Print to verify sorted


Reading C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\filtered128_folder\02010002erp 20150416 1131-epo.fif ...


    Found the data of interest:
        t =    -200.00 ...     800.00 ms
        0 CTF compensation matrices available
Not setting metadata
480 matching events found
No baseline correction applied
0 projection items activated
Overwriting existing file.
Overwriting existing file.
02010002erp 20150416 1131-epo.fif: ['E1', 'E2', 'E3', 'E4', 'E5', 'E6', 'E7', 'E8', 'E9', 'E10', 'E11', 'E12', 'E13', 'E14', 'E15', 'E16', 'E17', 'E18', 'E19', 'E20', 'E21', 'E22', 'E23', 'E24', 'E25', 'E26', 'E27', 'E28', 'E29', 'E30', 'E31', 'E32', 'E33', 'E34', 'E35', 'E36', 'E37', 'E38', 'E39', 'E40', 'E41', 'E42', 'E43', 'E44', 'E45', 'E46', 'E47', 'E48', 'E49', 'E50', 'E51', 'E52', 'E53', 'E54', 'E55', 'E56', 'E57', 'E58', 'E59', 'E60', 'E61', 'E62', 'E63', 'E64', 'E65', 'E66', 'E67', 'E68', 'E69', 'E70', 'E71', 'E72', 'E73', 'E74', 'E75', 'E76', 'E77', 'E78', 'E79', 'E80', 'E81', 'E82', 'E83', 'E84', 'E85', 'E86', 'E87', 'E88', 'E89', 'E90', 'E91', 'E92', 'E93', 'E94', 'E95', 'E96', 'E97', 'E98', 'E99', 

##### Checking if sorted channels are  in order 

In [ ]:
import mne
import os

# Path to your cleaned filtered folder
folder_path = r'C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\filtered128_folder'

# Loop through each .fif file
for file_name in os.listdir(folder_path):
    if file_name.endswith('.fif'):
        file_path = os.path.join(folder_path, file_name)
        try:
            raw = mne.read_epochs(file_path, preload=False, verbose=False)
            ch_names = raw.ch_names
            print(f"\n📁 {file_name}:")
            print(ch_names)
        except Exception as e:
            print(f"⚠️ Couldn't read {file_name}: {e}")



📁 02010002erp 20150416 1131-epo.fif:
['E1', 'E2', 'E3', 'E4', 'E5', 'E6', 'E7', 'E8', 'E9', 'E10', 'E11', 'E12', 'E13', 'E14', 'E15', 'E16', 'E17', 'E18', 'E19', 'E20', 'E21', 'E22', 'E23', 'E24', 'E25', 'E26', 'E27', 'E28', 'E29', 'E30', 'E31', 'E32', 'E33', 'E34', 'E35', 'E36', 'E37', 'E38', 'E39', 'E40', 'E41', 'E42', 'E43', 'E44', 'E45', 'E46', 'E47', 'E48', 'E49', 'E50', 'E51', 'E52', 'E53', 'E54', 'E55', 'E56', 'E57', 'E58', 'E59', 'E60', 'E61', 'E62', 'E63', 'E64', 'E65', 'E66', 'E67', 'E68', 'E69', 'E70', 'E71', 'E72', 'E73', 'E74', 'E75', 'E76', 'E77', 'E78', 'E79', 'E80', 'E81', 'E82', 'E83', 'E84', 'E85', 'E86', 'E87', 'E88', 'E89', 'E90', 'E91', 'E92', 'E93', 'E94', 'E95', 'E96', 'E97', 'E98', 'E99', 'E100', 'E101', 'E102', 'E103', 'E104', 'E105', 'E106', 'E107', 'E108', 'E109', 'E110', 'E111', 'E112', 'E113', 'E114', 'E115', 'E116', 'E117', 'E118', 'E119', 'E120', 'E121', 'E122', 'E123', 'E124', 'E125', 'E126', 'E127', 'E128']

📁 02010004erp 20141219 1602-epo.fif:
['E1', 

### Combining with metadata (that came along with the dataset)

In [ ]:
import os
import pandas as pd
import mne

# Path to EEG and metadata
eeg_folder = r"C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\filtered128_folder"
metadata_csv = r"C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\Pre_available_files\subjects_information_EEG_128channels_ERP_lanzhou_2015.xlsx"  # <- Save as CSV

# Load metadata
meta_df = pd.read_excel(metadata_csv, dtype={"Subject_Id": str})

# Ensure Subject_Id has consistent format (leading zeros)
meta_df['Subject_Id'] = meta_df['Subject_Id'].str.zfill(8)

# Map filenames to metadata
for fname in os.listdir(eeg_folder):
    if fname.endswith("-epo.fif"):
        subject_id = fname[:8]  # Extract from filename
        fpath = os.path.join(eeg_folder, fname)

        try:
            epochs = mne.read_epochs(fpath, preload=False, verbose=False)
            subj_meta = meta_df[meta_df['Subject_Id'] == subject_id]

            if subj_meta.empty:
                print(f"⚠️ No metadata for {subject_id}")
            else:
                print(f"✅ {fname} matched → {subj_meta.iloc[0].to_dict()}")
                
        except Exception as e:
            print(f"❌ Failed to read {fname}: {e}")


✅ 02010002erp 20150416 1131-epo.fif matched → {'Subject_Id': '02010002', 'type': 'MDD', 'age': 18, 'gender': 'F', 'education(years)': 12, 'PHQ-9': 23, 'CTQ-SF': 77, 'LES': -143, 'SSRS': 31, 'GAD-7': 18, 'PSQI': 12}
✅ 02010004erp 20141219 1602-epo.fif matched → {'Subject_Id': '02010004', 'type': 'MDD', 'age': 25, 'gender': 'F', 'education(years)': 19, 'PHQ-9': 12, 'CTQ-SF': 53, 'LES': -44, 'SSRS': 38, 'GAD-7': 13, 'PSQI': 11}
✅ 02010005erp 20150507 0938-epo.fif matched → {'Subject_Id': '02010005', 'type': 'MDD', 'age': 20, 'gender': 'M', 'education(years)': 16, 'PHQ-9': 19, 'CTQ-SF': 49, 'LES': -3, 'SSRS': 28, 'GAD-7': 11, 'PSQI': 5}
✅ 02010006erp 20150528 1007-epo.fif matched → {'Subject_Id': '02010006', 'type': 'MDD', 'age': 42, 'gender': 'M', 'education(years)': 16, 'PHQ-9': 16, 'CTQ-SF': 59, 'LES': -30, 'SSRS': 40, 'GAD-7': 12, 'PSQI': 9}
✅ 02010008_erp(n) 20150619 1709-epo.fif matched → {'Subject_Id': '02010008', 'type': 'MDD', 'age': 42, 'gender': 'M', 'education(years)': 12, 'PHQ

In [ ]:
import os
from collections import Counter
import mne
import numpy as np
fif_dir = r"C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\filtered128_folder"

channel_counts = []
epoch_counts = []

for fname in sorted(os.listdir(fif_dir)):
    if fname.endswith('-epo.fif'):
        filepath = os.path.join(fif_dir, fname)
        try:
            epochs = mne.read_epochs(filepath, preload=False)
            channel_counts.append(len(epochs.ch_names))
            epoch_counts.append(len(epochs))
        except Exception as e:
            print(f"{fname}: error - {e}")

print("Channel count distribution:", Counter(channel_counts))
print("Epoch count stats: min =", min(epoch_counts), "max =", max(epoch_counts), "avg =", np.mean(epoch_counts))


Reading C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\filtered128_folder\02010002erp 20150416 1131-epo.fif ...
    Found the data of interest:
        t =    -200.00 ...     800.00 ms
        0 CTF compensation matrices available
Not setting metadata
480 matching events found
No baseline correction applied
0 projection items activated
Reading C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\filtered128_folder\02010004erp 20141219 1602-epo.fif ...
    Found the data of interest:
        t =    -200.00 ...     800.00 ms
        0 CTF compensation matrices available
Not setting metadata
480 matching events found
No baseline correction applied
0 projection items activated
Reading C:\Modma_dataset\EEG_128channels_ERP_lanzhou_2015\EEG_128channels_ERP_lanzhou_2015\filtered128_folder\02010005erp 20150507 0938-epo.fif ...
    Found the data of interest:
        t =    -200.00 ...     800.00 ms
        0 CTF compensation matr